tests/core/data/test_excel_data_node.py

```diff
@@ -689,3 +693,170 @@ def test_read_write_after_modify_path(self):
         dn.write(read_data)
         for sheet, df in dn.read().items():
             assert np.array_equal(df.values, read_data[sheet].values)
+
+    def test_exposed_type_custom_class_after_modify_path(self):
+        path = os.path.join(pathlib.Path(__file__).parent.resolve(), "data_sample/example.xlsx")  # ["Sheet1", "Sheet2"]
+        new_path = os.path.join(
+            pathlib.Path(__file__).parent.resolve(), "data_sample/example_2.xlsx"
+        )  # ["Sheet1", "Sheet2", "Sheet3"]
+        dn = ExcelDataNode("foo", Scope.PIPELINE, properties={"default_path": path, "exposed_type": MyCustomObject1})
+        assert dn.exposed_type == MyCustomObject1
+        dn.read()
+        dn.path = new_path
+        dn.read()
+
+        dn = ExcelDataNode(
+            "foo",
+            Scope.PIPELINE,
+            properties={"default_path": path, "exposed_type": MyCustomObject1, "sheet_name": ["Sheet4"]},
+        )
+        assert dn.exposed_type == MyCustomObject1
+        with pytest.raises(NonExistingExcelSheet):
+            dn.read()
+
+    def test_exposed_type_dict_after_modify_path(self):
+        path = os.path.join(
+            pathlib.Path(__file__).parent.resolve(), "data_sample/example_2.xlsx"
+        )  # ["Sheet1", "Sheet2", "Sheet3"]
+        new_path = os.path.join(
+            pathlib.Path(__file__).parent.resolve(), "data_sample/example.xlsx"
+        )  # ["Sheet1", "Sheet2"]
+        dn = ExcelDataNode(
+            "foo",
+            Scope.PIPELINE,
+            properties={
+                "default_path": path,
+                "exposed_type": {
+                    "Sheet1": MyCustomObject1,
+                    "Sheet2": MyCustomObject1,
+                    "Sheet3": MyCustomObject1,
+                },
+            },
+        )
+        dn.read()
+        dn.path = new_path
+        dn.read()
+
+    def test_exposed_type_list_after_modify_path(self):
+        path_1 = os.path.join(
+            pathlib.Path(__file__).parent.resolve(), "data_sample/example.xlsx"
+        )  # ["Sheet1", "Sheet2"]
+        path_2 = os.path.join(
+            pathlib.Path(__file__).parent.resolve(), "data_sample/example_2.xlsx"
+        )  # ["Sheet1", "Sheet2", "Sheet3"]
+        path_3 = os.path.join(
+            pathlib.Path(__file__).parent.resolve(), "data_sample/example_3.xlsx"
+        )  # ["Sheet3", "Sheet4"]
+
+        dn = ExcelDataNode(
+            "foo", Scope.PIPELINE, properties={"default_path": path_1, "exposed_type": [MyCustomObject, MyCustomObject]}
+        )
+
+        dn.read()
+        dn.path = path_3
+        dn.read()
```

Comment:
I believe you don't need to test with 3 files. path_1 and path2 should be enough. (path_3 does not seem to be useful)